src/core/thread/router_table.cpp

```diff
@@ -342,21 +342,6 @@ void RouterTable::RemoveRouterLink(Router &aRouter)
     }
 }
 
-uint8_t RouterTable::GetActiveLinkCount(void) const
-{
-    uint8_t activeLinks = 0;
-
-    for (const Router *router = GetFirstEntry(); router != nullptr; router = GetNextEntry(router))
-    {
-        if (router->IsStateValid())
-        {
-            activeLinks++;
-        }
-    }
-
-    return activeLinks;
-}
-
```

Comment:
`GetNeighborCount()` does the same:
https://github.com/openthread/openthread/blob/4e4447143196192decdd8c4f101cb7ac991b8b12/src/core/thread/router_table.cpp#L452-L477